File: bookstoreapp291/lib/model/book.dart
import 'dart:io';

class Book {
  final String id;
  final String bookName;
  final String bookDescription;
  final int bookAmount;
  final String bookImage;

  Book.set(
      {required this.id,
      required this.bookName,
      required this.bookDescription,
      required this.bookAmount,
      required this.bookImage});

  factory Book.fromJson(Map<String, dynamic> json) {
    return Book.set(
        id: json['_id'],
        bookName: json['bookName'],
        bookDescription: json['bookDescription'],
        bookAmount: json['bookAmount'],
        bookImage: json['bookImage']);
  }

  Map<String, dynamic> toJson() {
    return {
      'bookName': bookName,
      'bookDescription': bookDescription,
      'bookAmount': bookAmount,
      'bookImage': bookImage
    };
  }
}

File: nodejs-express-template-main/src/models/book.ts
// const bmongoose = require('mongoose')
import mongoose, { Types } from 'mongoose'
// const bSchema = bmongoose.Schema

export const bookSchema = new mongoose.Schema({
    owner: {
        type: Types.ObjectId,
        required: true,
        ref: 'User'
    },
    bookName:{
        type: String, 
        require: true,
    },
    bookDescription:{
        type: String,
        require: true,
    },
    bookAmount:{
        type: Number,
        require: true
    },
    bookImage:{
        type: String,
        require: true
    }
})

export default mongoose.model('Book', bookSchema)